### DISCUSSION TITLE
How can I cause a push after a commit automatically

### DISCUSSION THREAD

**Discussion by rickprice** [tpope/vim-fugitive#1669] [Q&A] [answered]
I prefer to keep my source code pushed to my origin.

This means that I pretty much always have to do a :Gpush after my :Gcommit, which I find annoying.

Is there some way I can create a mapping in Vim, or could I create a PR that has a command named like CC (instead of cc) that would do the push as well?

I already tried to do a mapping, but it never worked, I'm assuming that is because it doesn't wait for the :wq on the :Gcommit.

Any help appreciated :).

Frederick

**Comment by tpope** [comment]
I don't think there is a clean way to do this inside Vim. Use a Git hook maybe?

**Comment by rickprice** [comment]
Right, but there are commands for cc inside :G, couldn't the :Gpush be
chained with that somehow?

Rick

**Comment by tpope** [comment, marked as answer]
`:Git commit` returns control to the user once the commit message is opened. So that's the only point you can easily chain off of.

**Reply by tpope** [reply]
Now that I think about it, you could create a *Git* alias that calls `git commit && git push` and call that with `:Git` and it will do what you expect.
